main.go

```diff
@@ -150,6 +147,33 @@ func main() {
 	}
 }
 
+func getConnectionState() *tls.ConnectionState {
```

Comment:
I'd prefer if this function took arguments instead of relying on the globals connectStartTLS and connectTo